docs/tutorials/using_datasets/IQL_torchrl.py

```diff
@@ -0,0 +1,522 @@
+# ~~~~~~~~
+
+# %%
+# To demonstrate training, we run IQL for 50,000 iterations. During training, we will evaluate the policy every 1000 iterations. But note that this is for evaluation purposes only. Unlike online RL, we do not collect new data during training.
+
+# %%
+@torch.no_grad()
+def evaluate_policy(env, policy, num_eval_episodes=20):
+    """Calculate the mean cumulative reward over multiple episodes."""
+    episode_rewards = []
+
+    for _ in range(num_eval_episodes):
+        eval_td = env.rollout(max_steps=max_episode_steps, policy=policy, auto_cast_to_device=True)
+        episode_rewards.append(eval_td["next", "reward"].sum().item())
+
+    return np.mean(episode_rewards)
+
+
+# %%
+# The training loop is essentially the standard PyTorch gradient descent loop:
+#
+# 1. Sample a batch of transitions from the dataset :math:`\mathcal{D}`.
+# 2. Compute the loss :math:`\ell = L_V(\psi) + L_Q(\theta) + L_\pi(\phi)`.
+# 3. Backpropagate the gradients and update the networks, including the target Q-network.
+
+# %%
+from tqdm.auto import tqdm
+
+iterations = 50_000
+eval_interval = 1_000
+
+loss_logs = []
+eval_reward_logs = []
+pbar = tqdm(range(iterations))
+
+for i in pbar:
+    # 1) Sample data from the dataset
+    data = replay_buffer.sample()
+
+    # 2) Compute loss l = L_V + L_Q + L_pi
+    loss_dict = loss_module(data.to(device))
+    loss = loss_dict["loss_value"] + loss_dict["loss_qvalue"] + loss_dict["loss_actor"]
+    loss_logs.append(loss.item())
+
+    # 3) Backpropagate the gradients
+    optimizer.zero_grad()
+    loss.backward()
+    optimizer.step()  # Update V(s), Q(a, s), pi(a|s)
+    target_net_updater.step()  # Update the target Q-network
+
+    # Evaluate the policy
+    if i % eval_interval == 0:
+        eval_reward_logs.append(evaluate_policy(env, model[0]))
+        pbar.set_description(
+            f"Loss: {loss_logs[-1]:.1f}, Avg return: {eval_reward_logs[-1]:.1f}"
+        )
+
+pbar.close()
+
+# %%
+# \
+# We can plot the loss and episodic return:
+
+# %%
+fig, axes = plt.subplots(nrows=1, ncols=2, figsize=(7, 3))
+axes[0].plot(loss_logs)
+axes[0].set_title("Loss")
+axes[0].set_xlabel("iterations")
+axes[1].plot(eval_interval * np.arange(len(eval_reward_logs)), eval_reward_logs)
+axes[1].set_title("Cumulative reward")
+axes[1].set_xlabel("iterations")
+fig.tight_layout()
+
+# %%
+# .. image:: /_static/img/tutorials/IQL_torchrl_training_graphs.png
+#    :width: 100%
+#    :align: center
+#    :alt: Loss and eval reward during training
+
+# %%
+# Results
+# ~~~~~~~
+
+# %%
+from IPython.display import HTML
+from gymnasium.utils.save_video import save_video
+from base64 import b64encode
+
+# %%
+# Evaluated over 100 episodes, the final performance is:
+
+# %%
+final_score = evaluate_policy(env, model[0], num_eval_episodes=100)
+print(f"Cumulative reward (averaged over 100 episodes): {final_score:.2f}")
+
+# %%
+# .. code-block:: text
+#
+#     Cumulative reward (averaged over 100 episodes): 1952.30
```

Comment:
as before, I get another value. Maybe, you run some cells more than one time (?)
Before merging we should run the whole notebook and fill in the correct values (and check reproducibility)

Reply:
Hopefully the seeding fix has addressed this as well. If not let me know!